src/lib/storage/frame_of_reference_segment.hpp

```diff
@@ -70,7 +70,7 @@ class FrameOfReferenceSegment : public BaseEncodedSegment {
 
   std::optional<T> get_typed_value(const ChunkOffset chunk_offset) const {
     // performance critical - not in cpp to help with inlining
-    if (_null_values[chunk_offset]) {
+    if (_null_values && (*_null_values)[chunk_offset]) {
```

Comment:
Theoretically, we could also add non-null iterator support to this now. Have a look at `ValueSegmentIterable::_on_with_iterators`. Just a suggestion, not a change request. We should also consider if the compile time costs justify the potential benefits.

Reply:
To be honest, that looks horrible to me. Shouldn't that be possible with a simple template parameter?
In any case, I would rather only use it for Dictionary and ValueSegments (especially with the increasing TPC-C importance). We just do not use FoR often enough to justify the compile time IMO.

But if you want I can take a look into the two main segments (there should be further improvements possible for value segments as far as I can see) and check if a templated version is possible. But probably only worth it for ValueSegments, right?

Reply:
I am not even sure if it has any benefits for value segments. Dictionary segments do not use that optimization, but it would be easy to add. If you are interested in this, you could simply disable the `is_null` check in the two dictionary iterators and run TPC-H.

Reply:
I'll take a look at it in another branch.